CHANGELOG.md

```diff
@@ -54,6 +54,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Changed the default of `find_unused_parameters` back to `True` in DDP and DDP Spawn ([#6438](https://github.com/PyTorchLightning/pytorch-lightning/pull/6438))
 
 
+- Changed the behavior of `on_epoch_start` to run at the beginning of validation & test epoch ([#6498](https://github.com/PyTorchLightning/pytorch-lightning/pull/6498))
```

Comment:
on the beginning of each? if so how `on_epoch_start` recognize which train/val/test it is?

Reply:
Why does it need to? For them we already have on_train/val/test_epoch_start/end separately. This one is just a common one. So if someone wants to do something in case of all train/val/test then they can use this one else we have separate hooks for each. Similarly we already have this for on_epoch_end.